tempodb/backend/backend.go

```diff
@@ -64,4 +79,6 @@ type Compactor interface {
 	ClearBlock(blockID uuid.UUID, tenantID string) error
 	// CompactedBlockMeta returns the compacted blockmeta given a block and tenant id
 	CompactedBlockMeta(blockID uuid.UUID, tenantID string) (*CompactedBlockMeta, error)
+	// DeleteTenantIndex deletes the tenant index.
+	DeleteTenantIndex(ctx context.Context, tenantID string) error
```

Comment:
Have an idea that could simplify things - what if this was done automatically inside `WriteTenantIndex`?  If there are no blocks then delete.